### PR TITLE
[Refactor:CourseMaterials] Use ZipStream for bulk downloading

### LINKED ISSUE #6229: Use zipstream for course materials download folder
As noted in PR #6214 

Instead of writing a (potentially large) zip file to /tmp, let's use the zip stream library for course materials download folder...  like we do in other parts of the site.

### PR DESCRIPTION
**Pull request by loadedskiffer**
### What is the current behavior?
closes #6229 

### What is the new behavior?
Course materials zip download now uses the ZipStream library 

### Other information?
To test:
As instructor upload folders to course materials
As student download the folders

### CODE REVIEW

**Review by cjreed121 — APPROVED** [review]

File: site/app/controllers/course/CourseMaterialsController.php [hunk 1]
```diff
@@ -129,13 +133,14 @@ public function downloadCourseMaterialZip($dir_name, $path) {
...
         }
-        $zip->close();
+        $zip_stream->finish();
         header("Content-type: application/zip");
         header("Content-Disposition: attachment; filename=$zip_file_name");
         header("Content-length: " . filesize($zip_name));
         header("Pragma: no-cache");
         header("Expires: 0");
         readfile("$zip_name");
+        unlink($zip_name);
```
Comment (MasterOdin) [inline site/app/controllers/course/CourseMaterialsController.php:143]: Is this already handled by `register_shutdown_function('unlink', $zip_name);` on line 92?

File: site/public/js/server.js [hunk 2]
```diff
@@ -659,6 +659,7 @@ function check_server(url) {
 }
 
 function downloadFile(path, dir) {
+    console.log(['download'] + `?dir=${encodeURIComponent(dir)}&path=${encodeURIComponent(path)}`);
```
Comment (MasterOdin) [inline site/public/js/server.js:662]: can be removed?

File: site/app/controllers/course/CourseMaterialsController.php (hunk 1 above)
Comment (loadedskiffer) [reply site/app/controllers/course/CourseMaterialsController.php]: I realized lines 137-143 are unnecessary when using zip stream and were causing problems with some downloads, they have been removed

**Review by MasterOdin — APPROVED** [review]
Could you also submit a PR to cleanup:

https://github.com/Submitty/Submitty/blob/01d292e2b854ce3a9a98e24d70919c17af43e1e7/site/app/controllers/MiscController.php#L352-L358?

**Pull request merged**